ahnlich/ai/src/tests/aiproxy_test.rs

```diff
@@ -623,6 +624,179 @@ async fn test_ai_proxy_get_sim_n_succeeds() {
+            }
+        } else {
+            MetadataValue {
+                value: Some(MValue::RawString("".into())),
+            }
+        };
+
+        let store_value = StoreValue {
+            value: HashMap::from_iter([(
+                matching_metadatakey.clone(),
+                matching_metadatavalue.clone(),
+            )]),
+        };
+
+        let store_input_1 = if index_model_repr.input_type().as_str_name() == "RAW_STRING" {
+            StoreInput {
+                value: Some(Value::RawString("Jordan 3".into())),
+            }
+        } else if index_model_repr.input_type().as_str_name() == "IMAGE" {
+            StoreInput {
+                value: Some(Value::Image(include_bytes!("./images/cat.png").to_vec())),
+            }
+        } else {
+            StoreInput {
+                value: Some(Value::RawString("".into())),
+            }
+        };
+
+        let store_input_2 = if index_model_repr.input_type().as_str_name() == "RAW_STRING" {
+            StoreInput {
+                value: Some(Value::RawString("Air Force 1".into())),
+            }
+        } else if index_model_repr.input_type().as_str_name() == "IMAGE" {
+            StoreInput {
+                value: Some(Value::Image(include_bytes!("./images/dog.jpg").to_vec())),
+            }
+        } else {
+            StoreInput {
+                value: Some(Value::RawString("".into())),
+            }
+        };
+
+        let store_data = vec![
+            AiStoreEntry {
+                key: Some(store_input_1.clone()),
+                value: Some(store_value.clone()),
+            },
+            AiStoreEntry {
+                key: Some(store_input_2.clone()),
+                value: Some(store_value.clone()),
+            },
+        ];
+
+        // Create pipeline request
+        let queries = vec![
+            ai_pipeline::AiQuery {
+                query: Some(Query::CreateStore(ai_query_types::CreateStore {
+                    store: store_name.clone(),
+                    query_model: model,
+                    index_model: model,
+                    predicates: vec![],
+                    non_linear_indices: vec![],
+                    error_if_exists: true,
+                    store_original: true,
+                })),
+            },
+            ai_pipeline::AiQuery {
+                query: Some(Query::Set(ai_query_types::Set {
+                    store: store_name.clone(),
+                    inputs: store_data,
+                    preprocess_action: PreprocessAction::NoPreprocessing.into(),
+                    execution_provider: None,
+                })),
+            },
+        ];
+
+        let pipelined_request = ai_pipeline::AiRequestPipeline { queries };
+
+        let _ = client
+            .pipeline(tonic::Request::new(pipelined_request))
+            .await
+            .expect("Failed to send pipeline request");
+
+        let expected_entries = ai_response_types::StoreInputToEmbeddingsList {
```

Comment:
If we switch to the assert like below we don't really need this then as we just need to clone each store_input to check that they exist in the response